wordpress-mobile/WordPress-Android
# Attach Purchase Domain screen to Domain Search flow

Fixes #19502

The PR adds the Purchase Domain screen to the existing flow. 

Note: I also converted `PurchaseDomainViewModel` to assisted injection-friendly implementation for passing domain id to the viewmodel's parameters, which makes the class easily testable. I wasn't sure if we needed the entire `ProposedDomain` entity for passing a selected domain through the further steps, but it should be easy to replace an ID with an entire `ProposedDomain` entity if needed.

To test:
- Make sure you have enabled domain_management feature flag
- Go to "Me" tab -> Domains -> Plus (+) button
- On the "Search for a domain" screen type any query to the text input
- Tap on any domain suggestion
- [x] Make sure you are navigated to the "Purchase Domain" screen

https://github.com/wordpress-mobile/WordPress-Android/assets/16563318/ed0a8b2a-d25b-4dd7-892c-f89323652f29

## Regression Notes
1. Potential unintended areas of impact
Purchase Domain screen since modified the ViewModel implementation

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:


UI Changes testing checklist:
